index.html

```diff
@@ -231,16 +231,44 @@ <h2>
         Interfacing with a payment handler
       </h2>
       <p>
-        The <dfn>steps to constrain a payment handler</dfn> with
+        The <dfn>steps to filter payment handlers</dfn> with
         <a>BasicCardRequest</a> <var>data</var> are given by the following
-        algorithm. If the end user inputs or selects a <a>card</a> that meets
-        the constraints of <var>data</var>, the algorithm returns a card as a
-        <a>BasicCardResponse</a>.
+        algorithm. If the payee invokes
+        <code>paymentRequest.canMakePayment()</code> or
+        <code>paymentRequest.show()</code>, the algorithm returns
+        <code>true</code> for matching cards. In case of
+        <code>canMakePayment()</code>, the output of this algorithm is returned
+        to the payee. In case of <code>show()</code>, the output of this
+        algorithm determines whether a payment handler is presented to the
+        user.
       </p>
       <ol class="algorithm" data-link-for="BasicCardResponse">
         <li data-link-for="BasicCardRequest">Let <var>requestedNetworks</var>
-        be the result of filtering <var>data</var>["<a>supportedNetworks</a>"]
-        for <a>networks</a> that are <a>known</a> by this payment handler.
+        be <var>data</var>["<a>supportedNetworks</a>"].
+        </li>
+        <li data-link-for="BasicCardRequest">Let <var>requestedTypes</var> be
+        <var>data</var>["<a>supportedTypes</a>"].
+        </li>
+        <li>If the <a>primary account number</a> is not from one of the
```

Comment:
We should say:

> For each <var>card</var> of the user's <a>cards</a>: 
>  1. if <var>card</var>'s <a>type</a> ...
>  1. if <var>card</var>'s <a>network</a>...

We don't need to talk about the "primary account number" because the concept of "network" should state that it's derived from the "primary account number"'s BIN.

Reply:
Could you explain why it would be preferable to define this algorithm as an iteration over a list of cards (instruments) instead of the list of operations for a single card (instrument)?

Reply:
either ways is fine... basically:

```
cards.filter(card => stepsToCheckInstrumentSupport(card))
```

However, "cards" needs to come from Basic Card Payment Handler..